sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -997,11 +997,14 @@ object EliminateSorts extends Rule[LogicalPlan] {
       case _ => false
     }
 
-    aggs.flatMap { e =>
-      e.collect {
-        case ae: AggregateExpression => ae.aggregateFunction
-      }
-    }.forall(isOrderIrrelevantAggFunction)
+    def checkValidAggregateExpression(expr: Expression): Boolean = expr match {
```

Comment:
nit: private?

Reply:
We cannot make it private because this is a nested function within `private isOrderIrrelevantAggs`

Reply:
Ur, this is an inner func. I see.